ci/prow/e2e-incluster-build

```diff
@@ -5,38 +5,34 @@ set -euxo pipefail
 echo "Deploy KMMO..."
 make deploy
 
-echo "Wait until the KMMO Deployment is Available"
-timeout 1m kubectl wait --for condition=Available deployments.apps -n kmm-operator-system kmm-operator-controller-manager 
-
 echo "Create a build secret..."
 kubectl create secret generic build-secret --from-literal=ci-build-secret=super-secret-value
+
 echo "Add a configmap that contain the kernel module build dockerfile..."
 kubectl apply -f ci/kmm-kmod-dockerfile.yaml
 
 echo "Add an kmm-ci Module that contains a valid mapping..."
-sed -e "s/KVER_CHANGEME/$(uname -r)/g" ci/module-kmm-ci-build.template.yaml | tee module-kmm-ci.yaml
+sed -e "s/KVER_CHANGEME/$(minikube ssh -- uname -r)/g" ci/module-kmm-ci-build.template.yaml | tee module-kmm-ci.yaml
 kubectl apply -f module-kmm-ci.yaml
 
-echo "Wait for the job to be created..."
-timeout 1m bash -c 'until kubectl get job -l kmm.node.kubernetes.io/module.name | grep kmm; do sleep 1; done'
+# Wait for the build pod to be created. `kubectl wait` doesn't support such option,
+# see https://github.com/kubernetes/kubernetes/issues/83242.
+echo "Waiting for the build pod to be created..."
+timeout 1m bash -c 'until kubectl get pods -o json | jq -er ".items[].metadata.name | select(.? | match(\"build\"))"; do sleep 1; done'
+POD_NAME=$(kubectl get pods -o json | jq -r '.items[].metadata.name | select(.? | match("build"))')
 
-echo "Check that the module gets loaded on the node..."
-timeout 1m bash -c 'until minikube ssh -- lsmod | grep kmm_ci_a; do sleep 3; done'
+echo "Check that the build secret is mounted to the build pod"
```

Comment:
why do we need this check?

Reply:
This was already in the test but basically, when we have a `build` section in the CR, we may have a `build` secret as well.

The test is also making sure the build secret is correctly mounted to the build job/pod.

Reply:
After chatting, the job is not really using that build secret in the CR, it would use it if the `minikube`  registry was protected by authZ but it isn't.

There is no reason to check if a secret was mounted in a flow in which we are not using that secret (this is more of a UT flow).

Removing that check.